backend/src/main/scala/com/softwaremill/bootzooka/http/model/headers/headers.scala

```diff
@@ -0,0 +1,86 @@
+package com.softwaremill.bootzooka.http.model.headers
+
+import akka.http.scaladsl.model.headers.{ModeledCustomHeader, ModeledCustomHeaderCompanion}
+
+import scala.util.Try
+
+private[headers] sealed trait HeaderDirective[H] {
+  def value: String
+}
+
+private[headers] sealed trait RequestHeaderDirective[H] extends HeaderDirective[H]
+
+private[headers] sealed trait ResponseHeaderDirective[H] extends HeaderDirective[H]
```

Comment:
The purpose was to model the HTTP. They are private because there is currently no need for them to be public.